xcode/Subconscious/Shared/Components/Common/Profile/FollowUserFormSheet.swift

```diff
@@ -147,15 +114,174 @@ struct FollowUserView: View {
     }
 }
 
-struct FollowUserView_Previews: PreviewProvider {
+struct FollowNewUserFormSheetView_Previews: PreviewProvider {
     static var previews: some View {
-        FollowUserView(
-            state: AddressBookModel(),
-            send: { action in }
+        FollowNewUserFormSheetView(
+            state: FollowNewUserFormSheetModel(),
+            send: { action in },
+            did: Did("did:key:123")!,
+            onAttemptFollow: {},
+            onCancel: {},
+            onDismissFailFollowError: {}
         )
     }
 }
 
+// MARK: Actions
+enum FollowNewUserFormSheetAction {
+    case form(FollowUserFormAction)
+    
+    case presentQRCodeScanner(_ isPresented: Bool)
+    case qrCodeScanned(scannedContent: String)
+    case qrCodeScanError(error: String)
+}
+
+typealias FollowNewUserFormSheetEnvironment = Void
+
+// MARK: Model
+struct FollowNewUserFormSheetModel: ModelProtocol {
```

Comment:
This name is a little verbose, open to renaming this subdomain. 

It's the sheet containing the form for adding a user manually by entering a DID and petname.